R/bundle.R

```diff
@@ -444,9 +444,21 @@ inferAppMode <- function(appDir, appPrimaryDoc, files) {
     return("shiny")
   }
 
-  # shiny directory
-  shinyFiles <- grep("^(server|app).r$", files, ignore.case = TRUE, perl = TRUE)
-  if (length(shinyFiles) > 0) {
+  # Shiny application or ShinyRmd using server.R
```

Comment:
It might be a bit more complicated that that. ui.Rmd is really just a convention (as is index.Rmd). You could also have foo.Rmd with a server.R. The real heuristic for rmd-shiny is that there is at least one .Rmd file in the directory with `runtime: shiny_prerendered` or `runtime: shinyrmd`. If that condition is met it's an "rmd-shiny" (no matter whether it has a server.R and no matter what the name of the .Rmd file containing the requisite `runtime` is).

Reply:
Ok. Thanks. This is helpful. That suggests a different precedence than we have today, which prefers running the directory as a Shiny application. Will need to do a little more digging. Looking at the runtime field is easily done here. The heuristic in Connect doesn't read the YAML today.